Gems/Multiplayer/Code/Tests/ClientHierarchyTests.cpp

```diff
@@ -0,0 +1,379 @@
+        {
+            ClientSimpleHierarchyTests::SetUp();
+
+            m_childOfChildEntity = AZStd::make_unique<AZ::Entity>(AZ::EntityId(3), "child of child");
+            m_childOfChildEntityInfo = AZStd::make_unique<EntityInfo>(*m_childOfChildEntity.get(), ChildOfChildNetEntityId, EntityInfo::Role::Child);
+
+            CreateDeepHierarchyOnClient(*m_childOfChildEntityInfo);
+
+            m_childOfChildEntity->FindComponent<AzFramework::TransformComponent>()->SetParent(m_childEntity->GetId());
+        }
+
+        void TearDown() override
+        {
+            m_childOfChildEntityInfo.reset();
+            StopAndDeleteEntity(m_childOfChildEntity);
+
+            ClientSimpleHierarchyTests::TearDown();
+        }
+
+        void CreateDeepHierarchyOnClient(EntityInfo& childOfChild)
+        {
+            PopulateHierarchicalEntity(childOfChild);
+            SetupEntity(childOfChild.m_entity, childOfChild.m_netId, NetEntityRole::Client);
+
+            // we need a parent-id value to be present in NetworkTransformComponent (which is in client mode and doesn't have a controller)
+            SetParentIdOnNetworkTransform(childOfChild.m_entity, m_childEntityInfo->m_netId);
+            SetHierarchyRootFieldOnNetworkHierarchyChild<NetworkHierarchyChildComponent>(childOfChild.m_entity, m_rootEntityInfo->m_netId);
+
+            // Create an entity replicator for the child entity
+            const NetworkEntityHandle childOfChildHandle(&childOfChild.m_entity, m_networkEntityTracker.get());
+            childOfChild.m_replicator = AZStd::make_unique<EntityReplicator>(*m_entityReplicationManager, m_mockConnection.get(), NetEntityRole::Authority, childOfChildHandle);
+            childOfChild.m_replicator->Initialize(childOfChildHandle);
+
+            childOfChild.m_entity.Activate();
+        }
+
+        AZStd::unique_ptr<AZ::Entity> m_childOfChildEntity;
+        AZStd::unique_ptr<EntityInfo> m_childOfChildEntityInfo;
+    };
+
+    TEST_F(ClientDeepHierarchyTests, Client_Activates_Hierarchy_From_Network_Fields)
+    {
+        EXPECT_EQ(
+            m_rootEntity->FindComponent<NetworkHierarchyRootComponent>()->GetHierarchyRoot(),
+            InvalidNetEntityId
+        );
+        EXPECT_EQ(
+            m_childEntity->FindComponent<NetworkHierarchyChildComponent>()->GetHierarchyRoot(),
+            RootNetEntityId
+        );
+        EXPECT_EQ(
+            m_childOfChildEntity->FindComponent<NetworkHierarchyChildComponent>()->GetHierarchyRoot(),
+            RootNetEntityId
+        );
+
+        EXPECT_EQ(
+            m_childEntity->FindComponent<NetworkHierarchyChildComponent>()->GetHierarchicalRoot(),
+            m_rootEntity.get()
+        );
+
+        EXPECT_EQ(
+            m_rootEntity->FindComponent<NetworkHierarchyRootComponent>()->GetHierarchicalEntities().size(),
+            3
+        );
+        if (m_rootEntity->FindComponent<NetworkHierarchyRootComponent>()->GetHierarchicalEntities().size() == 3)
```

Comment:
the if-statement required if we're already EXPECT_EQ in the line above?

Reply:
ignore the last comment, I see... forgot this will continue to execute even if some expectation fails.

Reply:
Yep.